common/overlay/overlay.go

```diff
@@ -34,6 +36,33 @@ type MountCfg struct {
 	Lbl string
 }
 
+func DirSupportsOverlay(path string) (bool, error) {
+	dir, err := os.OpenFile(path, syscall.O_RDONLY|syscall.O_DIRECTORY, 0755)
+	if err != nil {
+		return false, errwrap.Wrap(fmt.Errorf("cannot open %q", dir), err)
+	}
+	defer dir.Close()
+
+	readDirentBuf := make([]byte, 4096)
+	var n int
+	n, err = syscall.ReadDirent(int(dir.Fd()), readDirentBuf)
+	if err != nil {
+		return false, errwrap.Wrap(fmt.Errorf("cannot read directory %q", dir), err)
+	}
+	readDirentOffset := 0
+	for readDirentOffset < n {
+		dirent := (*syscall.Dirent)(unsafe.Pointer(&readDirentBuf[readDirentOffset]))
```

Comment:
Note to myself: This will crash starting from Go 1.7 (see https://github.com/golang/go/issues/16778) once we point outside of the allocated 4k memory block. As far as I see we are advancing by `int(dirent.Reclen)`, but not checking whether we hit the `readDirentBuf` capacity.